src/main/java/com/yelp/nrtsearch/server/luceneserver/StartIndexHandler.java

```diff
@@ -61,11 +63,12 @@ public StartIndexResponse handle(IndexState indexState, StartIndexRequest startI
       synchronized (shardState) {
         try {
           if (!shardState.isRestored()) {
+            RestoreIndex restoreIndex = startIndexRequest.getRestore();
             if (startIndexRequest.getRestore().getDeleteExistingData()) {
```

Comment:
nit: if you are going to move the local variable, you might as well use it on the next line.